subprojects/docs/src/docs/userguide/signingPlugin.xml

```diff
@@ -101,6 +101,84 @@ gradle.taskGraph.whenReady { taskGraph ->
         </section>
     </section>
 
+    <section id="sec:using_gpg_agent">
+        <title>Using gpg-agent</title>
+        <para>
+            By default the signing plugin uses a Java-based implementation of PGP for signing. This implementation cannot
+            use the gpg-agent program for managing private keys, though. If you want to use the gpg-agent, you can change
+            the signatory implementation used by the signing plugin to
+            <apilink class="org.gradle.plugins.signing.signatory.gnupg.GnupgSignatory" />.
+        </para>
+        <sample id="use-gnupg" dir="signing/gnupg-signatory" title="Sign with GnuPG">
+            <sourcefile file="build.gradle" snippet="configure-signatory" />
+        </sample>
+        <para>
+            This tells the signing plugin to use the <literal>GnupgSignatory</literal>
+            instead of the default <apilink class="org.gradle.plugins.signing.signatory.pgp.PgpSignatory" />.
+            The <literal>GnupgSignatory</literal> relies on the gpg2 progam to sign the artifacts. Of course, this requires
+            that GnuPG is installed.
```

Comment:
Does Bouncy Castle maybe provide an implementation for GnuPG we could use. It somewhat looks like it: https://github.com/weiliatgithub/bouncycastle-gpg-example/. The we don't have to call out to an external process.

Reply:
I agree that it would be good if the signing could be done without relying on external processes. However, I could not find a way to integrate with gpg-agent without invoking the actual gpg executable.